### PR TITLE
PST Tipping

### PR DESCRIPTION
**Pull request by t8**
This PR introduces PST tipping functionality to twig.

This code needs to be tested before this PR can be merged.

CC @divy-work

### CODE REVIEW

**Comment by t8** [comment]
A fair bit of this is spaghetti code, and I wasn't completely sure where to interface with the Arweave connection. Feel free to change my code around. Let me know when a good time to test this would be for you.

**Comment by divy-work** [comment]
Sure. Sounds fun!

File: src/arweave.ts [hunk 1]
```diff
@@ -83,9 +97,230 @@ export async function save(
...
+      expr1: "App-Name",
+      expr2: "nest.land",
+    },
+    expr2: {
+      op: "equals",
+      expr1: "Previous-TX",
+      expr2: currentTX.id,
+    },
+  };
+  const response = await arweaveInit.api.post(`arql`, successorsQuery);
```
Comment (johnletey) [inline src/arweave.ts:251]: @t8 This query can be simplified tremendously by using [`arql-ops`](https://github.com/aidanok/arql-ops) 😄

Comment (divy-work) [reply src/arweave.ts:251]: @johnletey Yes! I'm working on using that rn. :smile:

**Comment by divy-work** [comment]
Did a few tests without providing the wallet from `yolk` through the `api`.

Works perfect but without providing the wallet.

image

**Comment by divy-work** [comment]
@t8 After doing multiple tests by providing the wallet. I get `Unhandled Promise Rejection` in `Transaction.get` (defined in arweave-js). No idea how to overcome that error.

**Comment by t8** [comment]
@divy-work I didn't have time to test this today, but I would expect a similar error since we haven't made the PST contract yet. I'll work on making a test contract tomorrow and fully testing the functionality.

**Comment by divy-work** [comment]
No problem! I'll investigate more on this today.

**Comment by t8** [comment]
Yes, CommunityJS is the way to go for this 👍

**Pull request merged**